### ISSUE TITLE
Half day attendance and leaves are not covering all possible cases.

### ISSUE THREAD

**Issue by nikhilponnuru** [frappe/erpnext#25663]
## Description of the issue

Consider the following cases which ERPNext doesn't cover

1. A half day attendance is marked (via half day leave application) and for rest of the day employee checked in and out (say log in, out via biometric device).
2. Vice versa of above case.
3. A half day leave is taken but rest of the half day employee didn't work and hence is marked absent based on shift
4. Vice versa of above case. (i.e applied leave for second half half-day)

These cases cannot be recorded because ERPNext wont allow attendance to be marked if there is already a leave application (because of duplicity checks for attendance records) and vice versa.

## Context information (for bug reports)

**Output of `bench version`**
```
erpnext 13.2.0
frappe 13.2.0

```

## Steps to reproduce the issue

Steps as explained in the description.

### Observed result
All possible cases are not covered for half day attendance

### Expected result
All possible cases are to be covered for half day attendance

### Stacktrace / full error message

```
(paste here)
```

## Additional information

OS version / distribution, `ERPNext` install method, etc.

Mac OS, mojave.

**Comment by ChandrahasSK** [comment]
Agree, half day is not well handled. A lot of manual reconciliation is being done by HR wrt leave applications and employee checkin records in our implementation.

**Comment by pushkarjsh88** [comment]
Can we have more use cases for this ?

**Comment by pushkarjsh88** [comment]
Can we get more use cases pertaining to same?

**Comment by wojosc** [comment]
We have also run into this issue.

A similar case is, when "Mark Attendance" has been run and Employee Checkins are created after the Attendance has been marked.

Maybe it would be possible to have a kind of "Attendance Reconciliation Tool" over which Employee Checkins could be cleaned and matched even after the Shift has been marked.

We would need to give the "Attendance Reconciliation Tool" the power to change Attendance after Submit and recalculate hours.

Further Ideas on this?

**Comment by wojosc** [comment]
Hey @nikhilponnuru our solution to this for the moment is a addon app that we have created:
https://github.com/phamos-eu/HR-Addon

You'll find the documentation here:
https://doku.phamos.eu/books/hr-addon

Hope it might help :)
